src/checkstyle/fixThis.gradle

```diff
@@ -1,64 +1,51 @@
 task fixThis << {
 	fileTree("${buildDir}/reports/checkstyle").include('*.xml').each { report ->
-		println "processing $report"
 		def xml = new XmlParser(false, false).parse(report)
 		xml.file.each { f ->
-			// println "processing $f"
 			def errors = f.error
-			def hasThisError = false
 			def thisErrors = []
 			errors.each { error ->
 				if (error.@source == 'com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck') {
 					thisErrors.add(error)
-					hasThisError = true
 				}
 			}
-			// println hasThisError
-			// println f.@name
-			if (hasThisError) {
+			if (thisErrors) {
 				def errorInx = 0
 				def error = thisErrors[errorInx++]
-				def lx = Integer.valueOf(error.@line)
-				println error
 				def file = new File(f.@name)
+				println "Fixing file $file ..."
+				boolean headerFixed
 				def outSource = ''
-				def ln = 0
-				file.eachLine { line ->
-					// println line
-					ln++
-					def matcher = line =~ /Copyright (20\d\d)(?:-20\d\d)?/
-					if (matcher.count) {
-						def years = matcher[0][1]
-						if (years != now) {
-							years = years + "-$now"
-						}
-						line.replaceFirst(/(20\d\d)(?:-20\d\d)?/, years)
-					}
-					if (error && ln == lx) {
-						// println line
-						def beforeIndex = Integer.valueOf(error.@column) - 1
-						def chars = line.toCharArray()
-						for (int i = 0; i < beforeIndex; i++) {
-							if (chars[i] == '\t') { // tabs before code == 8
-								beforeIndex -= 7;
-							}
-							else if (chars[i] != ' ') { // tabs after code start are only counted as 1
-								break;
+				file.eachLine { line, ln ->
+					if (!headerFixed) {
+						def matcher = line =~ /Copyright (20\d\d)(?:-(20\d\d))?/
+						if (matcher.count) {
+							def year1 = matcher[0][1]
+							if (now != year1) {
+								if (now != matcher[0][2]) {
+									line = line.replaceFirst(/(20\d\d)(?:-20\d\d)?/, year1 + "-$now")
+								}
 							}
+							headerFixed = true
 						}
-						line = line.substring(0, beforeIndex) + "this." + line.substring(beforeIndex)
-						// println line
+					}
+
+					while (error && ln == (error.@line as int)) {
+						def message = error.@message
+						def property = message.substring(message.indexOf('\'') + 1, message.lastIndexOf('\''))
+						def thisPrefix = message.substring(message.indexOf('"') + 1, message.lastIndexOf('"'))
+
+						line = line.replaceFirst(/(.+)(?<!\.)${property}(\W.+)/, "\$1$thisPrefix$property\$2")
```

Comment:
I am not sure this will work in all cases - that's why I switched to using the columns rather than `replace` and only fixed one occurrence in each pass.

consider 

    " name = " + name